--- sandbox/sandbox.2c9007f	2026-05-04 11:31:09.040341447 +0200
+++ bwrap-sandbox	2026-05-07 13:54:18.053717383 +0200
@@ -35,7 +35,7 @@
 import pwd
 import sepolicy
 
-SEUNSHARE = "/usr/sbin/seunshare"
+BWRAP = "/usr/bin/bwrap"
 SANDBOXSH = "/usr/share/sandbox/sandboxX.sh"
 PROGNAME = "selinux-sandbox"
 try:
@@ -216,10 +216,10 @@
             if not self.__options.homedir or not self.__options.tmpdir:
                 self.usage(_("Homedir and tempdir required for level mounts"))
 
-        if not os.path.exists(SEUNSHARE):
+        if not os.path.exists(BWRAP):
             raise ValueError(_("""
 %s is required for the action you want to perform.
-""") % SEUNSHARE)
+""") % BWRAP)
 
     def __mount_callback(self, option, opt, value, parser):
         self.__mount = True
@@ -227,10 +227,10 @@
     def __x_callback(self, option, opt, value, parser):
         self.__mount = True
         setattr(parser.values, option.dest, True)
-        if not os.path.exists(SEUNSHARE):
+        if not os.path.exists(BWRAP):
             raise ValueError(_("""
 %s is required for the action you want to perform.
-""") % SEUNSHARE)
+""") % BWRAP)
 
         if not os.path.exists(SANDBOXSH):
             raise ValueError(_("""
@@ -309,9 +309,9 @@
             pass
 
         usage = _("""
-sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-W windowmanager ] [ -w windowsize ] [[-i file ] ...] [ -t type ] command
+bwrap-sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-W windowmanager ] [ -w windowsize ] [[-i file ] ...] [ -t type ] command
 
-sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-W windowmanager ] [ -w windowsize ] [[-i file ] ...] [ -t type ] -S
+bwrap-sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-W windowmanager ] [ -w windowsize ] [[-i file ] ...] [ -t type ] -S
 %s
 """) % types
 
@@ -473,11 +473,20 @@
 
     def __execute(self):
         try:
-            cmds = [SEUNSHARE, "-Z", self.__execcon]
+            cmds = [BWRAP, '--dev-bind', '/', '/', "--exec-label", self.__execcon]
             if self.__options.usecaps:
-                cmds.append('-C')
+                cmds += [ '--cap-add', 'all' ]
             if self.__mount:
-                cmds += ["-t", self.__tmpdir, "-h", self.__homedir, "-r", self.__runuserdir]
+                xdg_runtime_dir = '/run/user/%i' % os.getuid()
+                if 'XDG_RUNTIME_DIR' in os.environ:
+                     xdg_runtime_dir = os.environ['XDG_RUNTIME_DIR']
+
+                cmds += [
+                    "--bind", self.__tmpdir , '/tmp',
+                    "--bind", self.__tmpdir , '/var/tmp',
+                    "--bind", self.__homedir, os.path.expanduser("~"),
+                    "--bind", self.__runuserdir, xdg_runtime_dir,
+                ]
 
                 if self.__options.X_ind or self.__options.Y_ind:
                     if self.__options.dpi:
@@ -489,7 +498,11 @@
                         dpi = str(Gtk.Settings.get_default().props.gtk_xft_dpi / 1024)
 
                     if os.environ.get('WAYLAND_DISPLAY') is not None:
-                        cmds += ["-W", os.environ["WAYLAND_DISPLAY"]]
+                        cmds += ["--ro-bind-try", xdg_runtime_dir + '/' + os.environ["WAYLAND_DISPLAY"], xdg_runtime_dir + '/' + os.environ["WAYLAND_DISPLAY"]]
+                    # Passing pipewire is seunshare -P (not currently used upstream).
+                    cmds += ["--ro-bind-try", xdg_runtime_dir + '/pipewire-0', xdg_runtime_dir + '/pipewire-0']
+                    # Passing pulse is not supported upstream at all (yet often needed for audio).
+                    cmds += ["--ro-bind-try", xdg_runtime_dir + '/pulse', xdg_runtime_dir + '/pulse']
 
                     xmodmapfile = self.__homedir + "/.xmodmap"
                     xd = open(xmodmapfile, "w")
